src/pharmpy/modeling/basic_models.py

```diff
@@ -134,10 +148,92 @@ def create_basic_pk_model(
         if model.modelfit_results is not None
         else None,
     )
-    if modeltype == 'oral':
+
+    if (
+        modeltype == 'pk_oral'
+        or modeltype == 'pk_ivoral'
+        or (modeltype == 'basic_pk' and administration == 'oral')
+        or (modeltype == 'basic_pk' and administration == 'ivoral')
+    ):
         model = set_first_order_absorption(model)
         model = set_initial_estimates(model, {'POP_MAT': mat_init})
         model = add_iiv(model, list_of_parameters='MAT', expression='exp', initial_estimate=0.1)
+    if modeltype == 'pk_ivoral' or (modeltype == 'basic_pk' and administration == 'ivoral'):
+        # FIXME : Dependent on CMT column having 1 and 2 as values otherwise
+        # compartment structure don't match
+        model = add_bioavailability(model)
+        model = set_initial_estimates(model, {'POP_BIO': 0.5})
+
+        # Change BIO statement to BIO = LOG(POP_BIO / (1 - POP_BIO))
+        pop_bio = model.parameters["POP_BIO"]
+        BIO = sympy.Symbol("BIO")
+        model = model.replace(
+            statements=model.statements.reassign(
+                BIO, sympy.log(pop_bio.symbol / (1 - pop_bio.symbol))
+            )
+        )
+
+        # Add IIV to BIO
+        # TODO ? Should there be another initial estimate?
+        model = add_iiv(model, list_of_parameters="BIO", expression='add', initial_estimate=0.1)
+
+        # Change F statement to F = 1 / (1 + EXP(-BIO))
+        F = model.statements.ode_system.dosing_compartments[0].bioavailability
+        model = model.replace(statements=model.statements.reassign(F, 1 / (1 + sympy.exp(-BIO))))
+
+        # Set dosing to the CENTRAL compartment as well
+        ode = model.statements.ode_system
+        cb = CompartmentalSystemBuilder(ode)
+        if df is None:
+            central_dose = Bolus(sympy.Symbol('AMT'), admid=2)
```

Comment:
Don't assume the name `AMT`. If we have a datainfo the dosing column should be queried from there.

Reply:
To follow the same principle as other doses, I decided to use `dosing()` for this dose as well, so all doses are handled in the same way.